tests/unit/backend/test_gcp.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+from pytest import fixture, mark, raises
+from redis_om import NotFoundError
+
+from runner_manager.backend.gcloud import GCPBackend
+from runner_manager.models.backend import Backends, GCPConfig, GCPInstanceConfig
+from runner_manager.models.runner import Runner
+from runner_manager.models.runner_group import RunnerGroup
+
+
+@fixture()
+def gcp_group() -> RunnerGroup:
+    runner_group: RunnerGroup = RunnerGroup(
+        id=2,
+        name="test",
+        organization="test",
+        backend=GCPBackend(
+            name=Backends.gcloud,
+            config=GCPConfig(
+                project_id=os.getenv("GCP_PROJECT_ID"),
```

Comment:
You should follow the same env var as the gcloud cli for consistency and not invent your own:
`CLOUDSDK_CORE_PROJECT`
`CLOUDSDK_COMPUTE_ZONE`

This way setting those will also set the one interpreted by `gcloud`

Additionally setting those var as `Field` in the model could allow you to retrive those from the env directly:

```python

from pydantic import Field

class GCPConfig(BackendConfig):
    """Configuration for GCP backend."""

    project_id: str = Field(..., env="CLOUDSDK_CORE_PROJECT")
    zone: str = Field(..., env="CLOUDSDK_COMPUTE_ZONE")
    service_account_email: str = "default"


config = GCPConfig()
config.zone # should return the zone from the env var if set.
```

Reply:
my bad, then in this case let's change BackendConfig to inherit from BaseSettings in this case it will take env variable then you can do:
```python
class GCPConfig(BackendConfig):
    """Configuration for GCP backend."""

    project_id: str = Field(..., alias="CLOUDSDK_CORE_PROJECT")
    zone: str = Field(..., alias="CLOUDSDK_COMPUTE_ZONE")
    service_account_email: str = "default"
```